operations/helm/charts/mimir-distributed/templates/kafka/kafka-statefulset.yaml

```diff
@@ -0,0 +1,175 @@
+                  fieldPath: metadata.name
+            - name: KAFKA_CLUSTER_ID
+              value: "{{ .clusterId | default "" }}"
+            - name: KAFKA_NODE_ID
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.labels['apps.kubernetes.io/pod-index']
+            - name: KAFKA_PROCESS_ROLES
+              value: "broker,controller"
+            - name: KAFKA_LISTENERS
+              value: "PLAINTEXT://0.0.0.0:9092,CONTROLLER://0.0.0.0:9093"
+            - name: KAFKA_ADVERTISED_LISTENERS
+              value: PLAINTEXT://$(_POD_NAME).{{ include "mimir.resourceName" (dict "ctx" $ "component" "kafka") }}-headless.{{ $.Release.Namespace }}.svc:9092
+            - name: KAFKA_CONTROLLER_QUORUM_VOTERS
+              # Kafka fails to resolve DNS name with a trailing dot, e.g. "example.cluster.local." Thus, "trimSuffix" is here for the clusterDomain.
+              value: "{{ range $i := until (int $replicas) }}{{ $i }}@{{ include "mimir.resourceName" (dict "ctx" $ "component" "kafka") }}-{{ $i }}.{{ include "mimir.resourceName" (dict "ctx" $ "component" "kafka") }}-headless.{{ $.Release.Namespace }}.svc.{{ trimSuffix "." $.Values.global.clusterDomain }}:9093{{ if lt $i (sub (int $replicas) 1) }},{{ end }}{{ end }}"
+            - name: KAFKA_CONTROLLER_LISTENER_NAMES
+              value: "CONTROLLER"
+            - name: KAFKA_INTER_BROKER_LISTENER_NAME
+              value: "PLAINTEXT"
+            - name: KAFKA_LOG_DIRS
+              value: "/var/lib/kafka/data"
+              # The "REPLICATION_FACTOR" must be defined so the broker created offsets topic. Otherwise, the Mimir ingesters won't be able to start.
+            - name: KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR
+              value: "{{ min 3 $replicas }}"
+            - name: KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR
+              value: "{{ min 3 $replicas }}"
+            - name: KAFKA_TRANSACTION_STATE_LOG_MIN_ISR
+              value: "{{ min 2 $replicas }}"
+            - name: KAFKA_NUM_PARTITIONS
+              value: "100"
+            - name: KAFKA_AUTO_CREATE_TOPICS_ENABLE
+              value: "true"
+            {{- with .extraEnv }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          {{- with .extraEnvFrom }}
+          envFrom:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          resources:
+            {{- toYaml .resources | nindent 12 }}
+          securityContext:
+            {{- toYaml .containerSecurityContext | nindent 12 }}
+          volumeMounts:
+            {{- if .persistence.enabled }}
+            - name: kafka-data
+              mountPath: /var/lib/kafka
+            {{- end }}
+            - name: kafka-config
+              mountPath: /opt/kafka/config
+            - name: tmp
+              mountPath: /tmp
+            {{- with .extraVolumeMounts }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          livenessProbe:
```

Comment:
`livenessProbe` is a niche use case. I don't think we should set it here unless there are Kafka docs that recommend it.